sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -60,20 +90,21 @@ case class CreateViewCommand(
     child: LogicalPlan,
     allowExisting: Boolean,
     replace: Boolean,
-    isTemporary: Boolean)
+    viewType: ViewType)
   extends RunnableCommand {
 
   override protected def innerChildren: Seq[QueryPlan[_]] = Seq(child)
 
-  if (!isTemporary) {
-    require(originalText.isDefined,
-      "The table to created with CREATE VIEW must have 'originalText'.")
+  if (viewType == PermanentView) {
+    require(originalText.isDefined, "'originalText' must be provided to create permanent view")
   }
 
   if (allowExisting && replace) {
     throw new AnalysisException("CREATE VIEW with both IF NOT EXISTS and REPLACE is not allowed.")
   }
 
+  private def isTemporary = viewType == LocalTempView || viewType == GlobalTempView
+
```

Comment:
Having an `isTemporary` method in `ViewType` might be more convenient and helps in keeping consistency.